src/components/AqiGraph/AqiGraph.tsx

```diff
@@ -0,0 +1,311 @@
+ * unhealthy - array of points that fall in the "unhealthy for all" category
+ * veryUnhealthy - array of points that fall in the "very unhealthy" category
+ * hazardous - array of points that fall in the "hazardous" category
+ */
+interface GraphData {
+  good: Array<GraphElement>;
+  moderate: Array<GraphElement>;
+  sensitive: Array<GraphElement>;
+  unhealthy: Array<GraphElement>;
+  veryUnhealthy: Array<GraphElement>;
+  hazardous: Array<GraphElement>;
+}
+
+/**
+ * AQI Graph Display Component
+ * This component displays a graph for the last 24 hours of AQI data for the
+ * currently selected sensor.
+ */
+const AqiGraph: ({sensorDocId}: GraphProps) => JSX.Element = ({
+  sensorDocId,
+}: GraphProps) => {
+  const defaultYLimit = 300;
+  const hoursPerDay = 24;
+  const minutesPerHour = 60;
+  /* --------------- state maintenance variables ---------------  */
+  const [data, setData] = useState<GraphData>({
+    good: [],
+    moderate: [],
+    sensitive: [],
+    unhealthy: [],
+    veryUnhealthy: [],
+    hazardous: [],
+  });
+  const [yAxisLimit, setYAxisLimit] = useState(defaultYLimit);
+  const [yAxisTicks, setYAxisTicks] = useState<number[]>([]);
+  const [horizontalFill, setHorizontalFill] = useState<string[]>([]);
+
+  const {t} = useTranslation('graph');
+
+  useEffect(() => {
+    // Get last 24 hours AQI buffer from sensor doc
+    if (sensorDocId) {
+      const docRef = firestore.collection('sensors').doc(sensorDocId);
+      const good = 50; // Air quality is good (0-50)
+      const moderate = 100; // Air quality is acceptable (51-100)
+      const sensitiveGroups = 150; // Health risk for sensitive groups (101-150)
+      const unhealthy = 200; // Health risk for all individuals (151-200)
+      const veryUnhealthy = 300; // Very unhealthy for all individuals (201-300)
+
+      docRef.get().then(doc => {
+        if (doc.exists) {
+          const data = doc.data();
+          if (data) {
+            const aqiBuffer: Array<AqiBufferElement> = data.aqiBuffer ?? [];
+            const aqiBufferIndex: number = data.aqiBufferIndex ?? 0;
+            const allData: GraphData = {
+              good: [],
+              moderate: [],
+              sensitive: [],
+              unhealthy: [],
+              veryUnhealthy: [],
+              hazardous: [],
+            };
+
+            // Get the local time from the user's browser
+            const currentDate = new Date();
+            const currentDay = currentDate.getDay();
+            const currentHour =
+              currentDate.getHours() +
+              currentDate.getMinutes() / minutesPerHour;
+            let maxAqi = 0; // Keep track of the highest AQI value to scale the graph
+            for (let i = 0; i < aqiBuffer.length; i++) {
+              // Get the index starting at the oldest reading and wrapping
+              // once we reach the end of the buffer
+              const index = (aqiBufferIndex + i) % aqiBuffer.length;
```

Comment:
We could initialize a `bufferIndex` value outside of the loop and then just `++` the `bufferIndex` variable instead of repeatedly adding `i` to the `aqiBufferIndex`.

Reply:
I just switched to using the `i` from the for loop instead. This version is a legacy from the line graph where the order of the points is important.